studio/components/interfaces/Database/Triggers/CreateTrigger.tsx

```diff
@@ -256,28 +257,33 @@ interface CreateTriggerProps {
 }
 
 const CreateTrigger = ({ trigger, visible, setVisible }: CreateTriggerProps) => {
+  const { project } = useProjectContext()
   const { ui, meta } = useStore()
   const [isClosingPanel, setIsClosingPanel] = useState(false)
   const _localState = useLocalObservable(() => new CreateTriggerStore())
   _localState.meta = meta as any
 
-  // for the empty 'no tables' state link
-  const router = useRouter()
-  const { ref } = router.query
+  useTablesQuery(
+    {
+      projectRef: project?.ref,
+      connectionString: project?.connectionString,
+    },
+    {
+      onSuccess(tables) {
+        if (_localState.tables.length <= 0) {
```

Comment:
this looks wrong - is it supposed to be `>=`?

Reply:
oh never mind - think i understand, if localState has not been loaded, then override

Reply:
any reason why we need that logic though? why not just setTables either way?

Reply:
I think some of the stuff in `CreateTriggerStore` relies on indexes, so replacing the items could make things strange... I'd rather leave this as is for now and remove it when we refactor this. wdyt?